### PR TITLE
refactor(content): refactor content to be functional - FE-3079

### PR DESCRIPTION
**Pull request by dawidzarzycki**
### Proposed behaviour
Replace SCSS with StyledComponents 

### Current behaviour
SCSS

### Checklist

- [x] Commits follow our style guide
- [ ] <del> Screenshots are included in the PR if useful
- [ ] <del> All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [x] Storybook added or updated if required
- [ ] <del> Typescript `d.ts` file added or updated if required
- [ ] <del> Carbon implementation and Design System documentation are congruent

### CODE REVIEW

File: src/components/content/content.d.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import * as React from 'react';
 
 export interface ContentProps {
   align?: 'left' | 'center' | 'right';
-  as?: 'primary' | 'secondary';
+  variant?: 'primary' | 'secondary';
```
Comment (sjohnson-sage) [inline src/components/content/content.d.ts:5]: This needs to be a breaking change if we're renaming this prop - you can point them towards this codemod in the breaking change message: https://github.com/Sage/carbon-codemod/tree/master/transforms/rename-prop

Comment (sjohnson-sage) [reply src/components/content/content.d.ts:5]: Although it's out of scope for this deprecating SCSS task so maybe just leave it as `as` for now

File: src/components/content/content.style.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
+/* eslint-disable import/prefer-default-export */
+import styled, { css } from "styled-components";
+import { baseTheme } from "../../style/themes";
+
+export const StyledContent = styled.div`
+  & + & {
+    margin-top: 15px;
+  }
+
+  text-align: ${({ align }) => align && align};
```
Comment (sjohnson-sage) [inline src/components/content/content.style.js:10]:
```suggestion
${({ align }) => css`text-align: ${align};`
```

File: src/components/content/content.d.ts (hunk 1 above)
Comment (dawidzarzycki) [reply src/components/content/content.d.ts:5]: I would leave it but AS is a keyword for StyledSystem and it means different things ;/ 
https://styled-components.com/docs/api#as-polymorphic-prop

Comment (sjohnson-sage) [reply src/components/content/content.d.ts:5]: Ah I see, in that case the breaking change message will be fine 👍🏻

File: src/components/content/content.style.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+      ${variant === "secondary" &&
+      css`
+        color: ${theme.content.secondaryColor};
+        font-weight: normal;
+      `}
+    `;
+  }};
+`;
+
+export const StyledContentBody = styled.div`
```
Comment (sjohnson-sage) [inline src/components/content/content.style.js:38]: It would be better to export all of these at the bottom of the file, i.e. 
```export {
  StyledContent, 
  StyledContentTitle, 
  StyledContentBody
}````;

File: src/components/content/content.style.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+      ${align === "center" &&
+      inline &&
+      css`
+        width: 50%;
+      `}
+    `;
+  }};
+`;
+
+StyledContent.defaultProps = { theme: baseTheme };
```
Comment (sjohnson-sage) [inline src/components/content/content.style.js:73]: `StyledContent` and `StyledContentBody` don't actually use the theme so you don't need these lines for those two

**Comment by DipperTheDan** [comment]
I think you may need to update the cypress locators. They appear to be looking for elements that no longer exist and I think this is why your cypress tests are failing. 
Is this correct @DlgSHi ?

File: src/components/content/content.stories.mdx [hunk 5]
```diff
@@ -1,7 +1,7 @@
 import { Meta, Story, Preview, Props} from '@storybook/addon-docs/blocks';
-import Content from './content.js';
+import Content from './content.component.js';
 
-<Meta title="Content" parameters={{ info: { disable: true }}} />
+<Meta title="Design System/Content" parameters={{ info: { disable: true }}} />
```
Comment (DipperTheDan) [inline src/components/content/content.stories.mdx:4]: The Content component now appears in `Design System` and outside of the `Design System` in its own `Content` folder. Should we not have the test and the docs in one location so not to confuse consumers?

File: src/components/content/content.style.js [hunk 6]
```diff
@@ -0,0 +1,78 @@
+/* eslint-disable import/prefer-default-export */
```
Comment (sjohnson-sage) [inline src/components/content/content.style.js:1]: Why does this eslint lint rule need to be disabled here?

Comment (sjohnson-sage) [reply src/components/content/content.style.js:1]: I don't think it does

File: cypress/support/step-definitions/content-steps.js [hunk 7]
```diff
@@ -1,9 +1,7 @@
 import { contentBody, contentTitle, contentPreview } from '../../locators/content';
 
 const CONTENT_AS_LABEL = 'carbon-content--';
-const CONTENT_ALIGN_LABEL = `${CONTENT_AS_LABEL}align-`;
 const BODY_FULL_WIDTH_PARAMETR = `${CONTENT_AS_LABEL}body-full-width`;
```
Comment (DlgSHi) [inline cypress/support/step-definitions/content-steps.js:4]: do we still need this stuff ?

Comment (KatarzynaQA) [reply cypress/support/step-definitions/content-steps.js]: yes, but I've changed it. Instead of `const` I put it in step directly

**Review by DlgSHi — APPROVED** [review]
Approve for `cypress` stuff

**Review by DipperTheDan — DISMISSED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by DlgSHi — DISMISSED** [review]

**Review by DipperTheDan — DISMISSED** [review]

File: src/components/content/content.component.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+const Content = ({
+  variant = "primary",
+  children,
+  title,
+  inline = false,
+  align = "left",
+  titleWidth,
+  bodyFullWidth = false,
+}) => {
```
Comment (sjohnson-sage) [inline src/components/content/content.component.js:17]: Minor but saying as this component has no logic there's no need for the curly brackets and return statement, i.e.
```jsx
const Content = ({
  ...
}) => (
  <StyledContent>
  ...
  </StyledContent>
);
```

Comment (sjohnson-sage) [reply src/components/content/content.component.js:17]: You don't have to change this but just FYI

File: src/components/content/content.component.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+  return (
+    <StyledContent
+      align={align}
+      bodyFullWidth={bodyFullWidth}
+      data-component="content"
+    >
+      <StyledContentTitle
+        variant={variant}
+        inline={inline}
+        data-element="title"
```
Comment (sjohnson-sage) [inline src/components/content/content.component.js:27]: I think this should be more specific - `data-element="content-title"`

File: src/components/content/content.component.js [hunk 10]
```diff
@@ -0,0 +1,66 @@
...
+        titleWidth={titleWidth}
+        align={align}
+      >
+        {title}
+      </StyledContentTitle>
+
+      <StyledContentBody
+        variant={variant}
+        inline={inline}
+        data-element="body"
```
Comment (sjohnson-sage) [inline src/components/content/content.component.js:37]: Same as above `data-element="content-body"`

File: src/components/content/content.stories.mdx [hunk 11]
```diff
@@ -10,36 +10,28 @@ contrast ratio of at least 4.5.1, to meet the [WCAG 2.0 AA standard](https://www
...
 
-## Inline Content
+## Inline content
 <Preview>
-  <Story name="with Inline">
-    <Content title='Title' inline>
-      This is an example of some content
-    </Content>
+  <Story name="inline">
+    <Content title='Title' inline children='This is an example of some content'/>
```
Comment (sjohnson-sage) [inline src/components/content/content.stories.mdx:27]: This is a strange use of the `children` prop - why did you change that from just having the text as children normally?

File: src/components/content/content.stories.mdx [hunk 12]
```diff
@@ -10,36 +10,28 @@ contrast ratio of at least 4.5.1, to meet the [WCAG 2.0 AA standard](https://www
...
-
-## Secondary Styling
+## Secondary styling
 <Preview>
-  <Story name="with Secondary Styling">
-    <Content title='Title' as='secondary'>
-      This is an example of some content
-    </Content>
+  <Story name="secondary styling">
+    <Content title='Title' variant='secondary' children='This is an example of some content'/>
```
Comment (sjohnson-sage) [inline src/components/content/content.stories.mdx:34]: Same question as above and I think you should change it back

File: src/components/content/content.stories.mdx (hunk 11 above)
Comment (dawidzarzycki) [reply src/components/content/content.stories.mdx:27]: wait.... what is it :D Let me check it, I don't remember I did that

**Review by sjohnson-sage — APPROVED** [review]

**Review by DipperTheDan — DISMISSED** [review]

**Review by DipperTheDan — APPROVED** [review]

**Review by sjohnson-sage — APPROVED** [review]

**Review by DlgSHi — APPROVED** [review]
